lib/cocoapods-core/podfile/dsl.rb

```diff
@@ -273,32 +262,106 @@ def podspec(options = nil)
       #             pod 'SSZipArchive'
       #           end
       #
-      # @example  Defining an exclusive target
+      # @example  Defining a test target
       #
       #           target :ZipApp do
       #             pod 'SSZipArchive'
-      #             target :test, :exclusive => true do
+      #             target :ZipAppTests do
       #               pod 'JSONKit'
       #             end
       #           end
       #
       # @return   [void]
       #
-      def target(name, options = {})
-        if options && !options.keys.all? { |key| [:exclusive].include?(key) }
+      def target(name, options = nil)
+        if options
           raise Informative, "Unsupported options `#{options}` for " \
-            "target `#{name}`"
+            "target `#{name}`."
         end
 
         parent = current_target_definition
         definition = TargetDefinition.new(name, parent)
-        definition.exclusive = true if options[:exclusive]
         self.current_target_definition = definition
-        yield
+        yield if block_given?
       ensure
         self.current_target_definition = parent
       end
 
+      # Defines a new abstract target that can be used for convenient
+      # target inheritance.
+      #
+      # @param    [Symbol, String] name
+      #           the name of the target.
+      #
+      # @example  Defining an abstract target
+      #
+      #           abstract_target 'Networking' do
+      #             pod 'SSZipArchive'
+      #             target 'Networking App 1'
+      #             target 'Networking App 2'
+      #           end
+      #
+      # @return   [void]
+      #
+      def abstract_target(name)
+        target(name) do
+          abstract!
+          yield if block_given?
+        end
+      end
+
+      # Denotes that the current target is abstract, and thus will not directly
+      # link against an Xcode target.
+      #
+      # @return [void]
+      #
+      def abstract!(abstract = true)
+        current_target_definition.abstract = abstract
+      end
+
+      # Sets the inheritance mode for the current target.
+      #
+      # @param   [:complete, :none, :search_paths] the inheritance mode to set.
+      #
+      # @example Inheriting only search paths
+      #
+      #          target 'App' do
+      #            target 'AppTests' do
+      #              inherit! :search_paths
```

Comment:
I think the default makes sense, as the most common case is probably sharing Pods between multiple targets.